[#5156] Add dialog for splitting items in inventory into two stacks

arbron opened the pull request:
Adds a "Split Stack" context menu option for an item that has a quantity of at least 2. Contains a modified version of core's `<range-picker>` element that has a number input on the left side. ~~This modified element won't be required if
https://github.com/foundryvtt/foundryvtt/issues/12147 is resolved.~~

Closes #5156

arbron committed:
[#5156] Add dialog for splitting items in inventory into two stacks

Adds a "Split Stack" context menu option for an item that has a
quantity of at least 2. Contains a modified version of core's
`<range-picker>` element that has a number input on the left side.
This modified element won't be required if
https://github.com/foundryvtt/foundryvtt/issues/12147 is resolved.

Closes #5156

krbz999 commented on module/applications/item/split-stack-dialog.mjs:
```diff
@@ -0,0 +1,69 @@
+    const right = formData.object.right ?? 0;
+    const left = (this.options.document.system.quantity ?? 1) - right;
+    if ( left === this.options.document.system.quantity ) return;
+    await this.options.document.update({ "system.quantity": left }, { render: false });
```
Is `render: false` to prevent the actor sheet rendering twice? This would prevent the item sheet rendering too, though. So maybe just `Promise.all` like is done with advancements?

arbron committed:
[#5156] Switch to using `modifyBatch` for update

Fyorl commented on module/applications/item/split-stack-dialog.mjs:
```diff
@@ -0,0 +1,75 @@
+   * @param {HTMLFormElement} form       The submitted form.
+   * @param {FormDataExtended} formData  Data from the dialog.
+   */
+  static async #handleFormSubmission(event, form, formData) {
```
Could we factor out the internals to a method on `Item5e`? And then we call it here with: `doc.split(right)`. That would allow us a shortcut if quantity is 2. In the context menu split handler we can check `if ( quantity === 2 ) return item.split(1)`, otherwise we invoke this dialog. Not super necessary but some nice polish I suppose.

arbron committed:
[#5156] Move `split` into `PhysicalItemTemplate`, auto-split

Move the code for splitting into `PhysicalItemTemplate#split` and
if the original item only has a quantity of `2` then skip the
dialog and just split in half.
